dali/kernels/slice/slice_gpu.cuh

```diff
@@ -99,35 +100,29 @@ union PackedBuffer {
 
 /**
  * @brief Simplified algorithm when no padding is necessary
- * @remarks `in` already refers to the slice anchor start
  */
 template <int Dims, typename OutputType, typename InputType>
 __device__ void SliceFuncNoPad(OutputType *__restrict__ out, const InputType *__restrict__ in,
                                const fast_div<uint64_t> *out_strides, const int64_t *in_strides,
-                               uint64_t offset, uint64_t block_end) {
-  if (Dims > 1 && out_strides[Dims - 1] == static_cast<uint32_t>(in_strides[Dims - 1])) {
-    const int NextDims = Dims > 1 ? Dims - 1 : 1;
-    SliceFuncNoPad<NextDims, OutputType, InputType>(out, in, out_strides, in_strides, offset,
-                                                    block_end);
-    return;
-  }
-
+                               const int64_t *anchor, const int64_t *step, uint64_t offset,
+                               uint64_t block_end) {
```

Comment:
Can we restore this optimization as well, similarly to what was done fpr the general case?
```suggestion
                               uint64_t block_end) {
                               
  if (Dims > 1 &&  step[Dims - 1] == 1 && step[Dims - 2] == 1 && out_strides[Dims - 1] == static_cast<uint32_t>(in_strides[Dims - 1])) {
    const int NextDims = Dims > 1 ? Dims - 1 : 1;
    SliceFuncNoPad<NextDims, OutputType, InputType>(out, in, out_strides, in_strides, anchor, step, offset,
                                                    block_end);
    return;
  }
```

Reply:
Added back with `anchor[Dim -1] == 0` which was also needed, passes test_subscript, running operator_2 now.